### PR TITLE
[jit] Handling of pretty-printing methods

### PR DESCRIPTION
**Pull request by zdevito**
Stacked on #14176, review only the last commit.
* Print parameters to methods as self.weight rather than as extra inputs.
* Print entire set of methods out as a single string
* Update test code to test the module-at-a-time export/import

### CODE REVIEW

File: torch/csrc/jit/passes/python_print.cpp [hunk 1]
```diff
@@ -808,12 +901,36 @@ struct PythonPrintPass {
     }
   }
   void printMethod(script::Method& method) {
+    std::unordered_map<at::Tensor*, QualifiedNamePtr> parameter_names;;
+    createTensorToParameterNameMap(method.owner(), QualifiedName::create("self"),  parameter_names);
+    printOneMethod(method, parameter_names);
+  }
+  void printOneMethod(
```
Comment (suo) [inline torch/csrc/jit/passes/python_print.cpp]: I think the `printOneX` names are confusing.

File: torch/csrc/jit/script/module.h [hunk 2]
```diff
@@ -186,6 +189,12 @@ struct Method {
     return optimize;
   }
 
+  // lookup owning module, liftime is module is only valid for the lifetime
```
Comment (suo) [inline torch/csrc/jit/script/module.h]: type: lifetime (also the wording is confusing)

**Review by suo — APPROVED** [review]
no substantive comments; I left a bunch of style nits as a present

File: torch/csrc/jit/passes/python_print.cpp [hunk 3]
```diff
@@ -753,10 +829,19 @@ struct PythonPrintPass {
 
     // current graph is used to de-dup names within a single graph
     scanBlock(graph.block());
-    assignValuesToTheirUniqueNames(graph.inputs());
+
+    // last param_names.size() arguments to the graph are parameters and not
+    // actual inpputs, we will print these as, e.g. self.foo.bar
+    // while we print the true_inputs out as parameters
+    auto true_inputs = graph.inputs().slice(0, graph.inputs().size() - param_names.size());
```
Comment (suo) [inline torch/csrc/jit/passes/python_print.cpp]: This code feels brittle to me, but it looks like this matches the nn.Module api?

File: test/test_jit.py [hunk 4]
```diff
@@ -237,29 +237,32 @@ def disableModuleHook(self):
...
-                    print(method2.graph)
-                    print(pp2)
-                    self.assertMultiLineEqual(pp, pp2)
+            try:
+                pp, constant_table = module._python_print()
+            except RuntimeError as e:
+                if "could not export python function" not in str(e):
+                    raise
+                else:
+                    return
```
[the remark this answers is not in the document]
Comment (zdevito) [reply test/test_jit.py]: There are a lot of graphs that contain python function calls that we know we cannot export. We just skip those in the hook. Other errors will still propagate.

File: torch/csrc/jit/passes/python_print.cpp (hunk 3 above)
Comment (zdevito) [reply torch/csrc/jit/passes/python_print.cpp]: it matches how Method works. The contract here is that the last param_names inputs are considered parameters. If we change module then we will have to change this pretty printing code as well.

**Pull request closed without merging**